### DISCUSSION TITLE
Proper way to revalidate data when using infinite scrolling.

### DISCUSSION THREAD

**Discussion by samhithk** [TanStack/query#7917] [Q&A]
I have a view similar to the one shown below. I am having some difficulty figuring out the best approach to revalidating data on an update. The `ItemList` needs to render a bunch of items and make use of infinite scrolling. When clicking on a single item it renders the `ItemDetails` on the right. These details are editable and when the user hits save I need to invalidate the appropriate query to reflect the update both on the `ItemDetails` view as well as the single `Item` in the `ItemList`. From what I can gather in the docs it looks like the `useInfiniteQuery` hook is my best bet. However when the update occurs is it best to revalidate everything so all items are refreshed or only the single page in which the updated item resides? Not sure what the best way to do this is or if there is a better approach. Any advice is appreciated 🙏

![image](https://github.com/user-attachments/assets/b968b0ec-10e4-4566-82ff-bdb407275d57)

**Comment by TkDodo** [comment]
you cannot refetch a single page within an infinite query, so you have to either write directly to the cache or invalidate the whole list (all pages)
